CMakeLists.txt

```diff
@@ -26,21 +26,33 @@ MESSAGE(STATUS  "binder: BINDER_USE_PYTHON_IN_TEST=${BINDER_USE_PYTHON_IN_TEST}"
 MESSAGE(STATUS  "binder: Using LLVM_DIR=${LLVM_DIR} to search for cmake configurations.")
 MESSAGE(STATUS  "binder: Using Clang_DIR=${Clang_DIR} to search for cmake configurations.")
 MESSAGE(STATUS  "binder: Using CMAKE_MODULE_PATH=${CMAKE_MODULE_PATH} to search for cmake configurations.")
-add_custom_target(LLVMSupport)
 find_package(Clang CONFIG QUIET PATHS /usr/lib64/cmake/clang /usr/lib/cmake/clang /usr/share/clang/cmake )
 find_package(LLVM  CONFIG QUIET PATHS /usr/lib64/cmake/clang  /usr/lib/cmake/llvm  /usr/share/llvm/cmake )
 if (Clang_FOUND AND LLVM_FOUND AND NOT LLVMCONFIG )
   set(CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR};${CMAKE_MODULE_PATH}") 
   MESSAGE(STATUS  "binder: cmake configurations  llvm and clang were found.  LLVM_CMAKE_DIR=${LLVM_CMAKE_DIR}")
   include(AddLLVM)
   include(LLVMConfig)
-  IF(EXISTS ${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION}/include)
-    set(LibClang_INCLUDE_DIR ${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION}/include)
-  else()
-    IF(EXISTS ${LLVM_LIBRARY_DIR}/../clang/${LLVM_VERSION}/include)
-      set(LibClang_INCLUDE_DIR ${LLVM_LIBRARY_DIR}/../clang/${LLVM_VERSION}/include)
-    endif()
-  endif()
+  MESSAGE(STATUS  "binder: LLVM_LIBRARY_DIR=${LLVM_LIBRARY_DIR}")
+  MESSAGE(STATUS  "binder: LLVM_VERSION=${LLVM_VERSION}")
+  find_path(LibClang_INCLUDE_DIR stddef.h HINTS
+    ${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION}/include
+    ${LLVM_LIBRARY_DIR}/../clang/${LLVM_VERSION}/include
+    ${LLVM_LIBRARY_DIR}/../../lib/clang/${LLVM_VERSION}/include
+    ${LLVM_LIBRARY_DIR}/../../lib64/clang/${LLVM_VERSION}/include
+    ${CLANG_INCLUDE_DIRS}/clang/${LLVM_VERSION}/include
+    ${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
+    ${LLVM_LIBRARY_DIR}/../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
+    ${LLVM_LIBRARY_DIR}/../../lib/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
+    ${LLVM_LIBRARY_DIR}/../../lib64/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
+    ${CLANG_INCLUDE_DIRS}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
+    ${LLVM_LIBRARY_DIR}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
+    ${LLVM_LIBRARY_DIR}/../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
+    ${LLVM_LIBRARY_DIR}/../../lib/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
+    ${LLVM_LIBRARY_DIR}/../../lib64/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
+    ${CLANG_INCLUDE_DIRS}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
```

Comment:
This has became quite long. Could you please see if it would be possible to refactor this into for-loop iterating over list of `clang`, `../clang`, `../lib/clang/`, `../lib64/clang/` and second loop iterating over list of [`${LLVM_VERSION}`, `${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}`]? Thanks,

Reply:
Hi @lyskov ,
yes, this is ugly. But the uglies part 
is exactly the workaround for the bugged installation when the location of headers is /10.0.0/  despite the version is 10.0.1.

Unfortunately a loop would not make things easier.   A regexp would help, but it is not supported.
The closest thing would be something like
```
    HINTS
    ${LLVM_LIBRARY_DIR}/clang
    ${LLVM_LIBRARY_DIR}/../clang
    ${LLVM_LIBRARY_DIR}/../../lib/clang
    ${LLVM_LIBRARY_DIR}/../../lib64/clang
    ${CLANG_INCLUDE_DIRS}/clang          
    PATH_SUFFIXES
    ${LLVM_VERSION}/include
    ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include
    ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.0/include
```

but it is less explicit and I don't know if it will work.

Best regards,

Andrii

Reply:
aha, - i see your point. Lets use current solution then.